### PR TITLE
Separate LyvEffectiveModelContext funcionality

### PR DESCRIPTION
**Pull request by PeterSuna**
Separate LyvEffectiveModelContext and create dedicated LyvEffectiveModelContextFactory class with SchemaTree initialization

### CODE REVIEW

File: src/main/java/io/lighty/yang/validator/Main.java [hunk 1]
```diff
@@ -97,95 +87,63 @@ public static void main(final String[] args) {
...
-
-        stopWatch.stop();
-        LOG.debug("Elapsed time: {}", stopWatch);
+        final var checkUpdateFrom = new CheckUpdateFrom(lyvContext.context(), module, contextFrom,
+                config.getCheckUpdateFrom(), config.getCheckUpdateFromConfiguration().getRfcVersion());
+        checkUpdateFrom.validate();
+        checkUpdateFrom.printErrors();
     }
 
     private static void runLyvForProvidedFiles(final Configuration config,  final Format format)
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/Main.java:140]: ` private static void runLyvForProvidedFiles(final Configuration config, final Format format)`

File: src/main/java/io/lighty/yang/validator/LyvEffectiveModelContextFactory.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if (yangFiles.isEmpty() && config.getTreeConfiguration().isHelp()) {
+            return new LyvEffectiveModelContext(null, null);
+        }
+        final var contextFactory = new YangContextFactory(config.getPath(), yangFiles, config.getSupportedFeatures(),
+                config.isRecursive());
+        final EffectiveModelContext context;
+        try {
+            context = contextFactory.createContext(config.getSimplify() != null);
+        } catch (final IOException | YangParserException e) {
+            throw new LyvApplicationException("Failed to create SchemaContext", e);
```
Comment (ihrasko) [inline src/main/java/io/lighty/yang/validator/LyvEffectiveModelContextFactory.java:45]: `throw new LyvApplicationException("Failed to assemble Effective Model Context", e);`

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: src/main/java/io/lighty/yang/validator/Main.java (hunk 1 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/Main.java]: Done

File: src/main/java/io/lighty/yang/validator/LyvEffectiveModelContextFactory.java (hunk 2 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/LyvEffectiveModelContextFactory.java]: Done

**Pull request merged**